src/esm_runscripts/filedicts.py

```diff
@@ -158,6 +191,42 @@ def allowed_to_be_missing(self):
         """
         return self.get("allowed_to_be_missing", False)
 
+    @property
+    def datestamp_method(self):
+        """
+        Defines which datestamp_method shall be used when possibly including
+        date stamps to the file. Valid choices are "never", "always",
+        "avoid_overwrite".
+        """
+        datestamp_method = self.get(
+            "datestamp_method", "avoid_overwrite"
+        )  # This is the old default behaviour
+        return datestamp_method
+
+    @property
+    def datestamp_format(self):
+        """
+        Defines which datestamp_format shall be used when possibly including
+        date stamps to the file. Valid choices are "check_from_filename" and
+        "append".
+        """
+        datestamp_format = self.get(
+            "datestamp_format", "append"
+        )  # This is the old default behaviour
+        return datestamp_format
+
+    def update(self, *args, **kwargs):
+        """
+        Standard dictionary update method, enhanced by additional safe-guards
+        for particular values.
+        """
+        for k, v in dict(*args, **kwargs).items():
+            if k == "datestamp_format":
+                self._check_datestamp_format_is_allowed(v)
+            if k == "datestamp_method":
+                self._check_datestamp_method_is_allowed(v)
+            self[k] = v
```

Comment:
Is it okay to not use `super().update` here? I ask because I am lacking the knowledge on whether `self[k] = v` is identical to `self.update({k: v})`

Reply:
Short answer: it is. What update basically does is: if a key exists, replace it with this new value. If a key does not exist, add it with this new value. So it's basically the same.